test/e2e/common/expansion.go

```diff
@@ -659,9 +659,17 @@ func waitForPodContainerRestart(f *framework.Framework, pod *v1.Pod, volumeMount
 	framework.ExpectNoError(err, "while waiting for container to restart")
 
 	// Fix liveness probe
-	ginkgo.By("Rewriting the file")
-	stdout, _, err = f.ExecShellInPodWithFullOutput(pod.Name, fmt.Sprintf("echo test-after > %v", volumeMount))
-	framework.Logf("Pod exec output: %v", stdout)
+	ginkgo.By("Attempting to rewrite the file")
+
+	// We might be trying to exec while the container is down. We should retry a few times.
```

Comment:
Would it be more reliable to have 2 containers? One container that is subject to the liveness probe, and the 2nd container that is not? Then we can be confident that the 2nd container should never be restarted.

Reply:
I agree, it seems two containers is a cleaner solution that is more deterministic.

Reply:
Done.

Reply:
I'm not seeing the update, did you push?

Reply:
Oh I'm sorry. That's my bad, I forgot to push. :)